ui/src/services/hooks/api.ts

```diff
@@ -87,9 +88,15 @@ export const useLocalStack = (): useLocalStackReturn => {
     cacheKey,
     async () => (await ddClient.docker.listContainers() as [DockerContainer])
       .find(container =>
-        container.Image === 'localstack/localstack' &&
-        container.Command !== 'bin/localstack update docker-images',
-      ), { refreshInterval: 2000,  compare: (a, b) => a?.Id === b?.Id },
+        isALocalStackContainer(container) && container.Command !== 'bin/localstack update docker-images',
+      ), {
+      refreshInterval: 2000, compare:
+      /*
+       * The Status of a container has a string formed like "Up N seconds (healthy)" or "Up N seconds (unhealthy)"
+       * so x?.Status.split('(').at(1) returns "healthy)" or "unhealthy)" and we can detect when this change occurs
+       */
+      (a, b) => a?.Id === b?.Id && a?.Status.split('(').at(1) === b?.Status.split('(').at(1), 
```

Comment:
Thanks for adding the comment to explain what's going on!
How about changing this to check whether `healthy` is included in `a.Status` and `b.Status`?
I.e. `a?.Status.includes('unhealthy') === b?.Status.includes('unhealthy')`

comment could then say `compares whether the old (b) status aligns with that of new (a) status'